src/Analyzers/CSharp/CodeFixes/UseImplicitObjectCreation/CSharpUseImplicitObjectCreationCodeFixProvider.cs

```diff
@@ -5,72 +5,81 @@
+        => !diagnostic.IsSuppressed;
 
-        public override Task RegisterCodeFixesAsync(CodeFixContext context)
-        {
-            RegisterCodeFix(context, CSharpAnalyzersResources.Use_new, nameof(CSharpAnalyzersResources.Use_new));
-            return Task.CompletedTask;
-        }
+    public override Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        RegisterCodeFix(context, CSharpAnalyzersResources.Use_new, nameof(CSharpAnalyzersResources.Use_new));
+        return Task.CompletedTask;
+    }
 
-        protected override Task FixAllAsync(
-            Document document, ImmutableArray<Diagnostic> diagnostics,
-            SyntaxEditor editor, CodeActionOptionsProvider fallbackOptions, CancellationToken cancellationToken)
-        {
-            // process from inside->out so that outer rewrites see the effects of inner changes.
-            foreach (var diagnostic in diagnostics.OrderBy(d => d.Location.SourceSpan.End))
-                FixOne(editor, diagnostic, cancellationToken);
+    protected override async Task FixAllAsync(
+        Document document, ImmutableArray<Diagnostic> diagnostics,
+        SyntaxEditor editor, CodeActionOptionsProvider fallbackOptions, CancellationToken cancellationToken)
+    {
+        // process from inside->out so that outer rewrites see the effects of inner changes.
+        var nodes = diagnostics
+            .OrderBy(d => d.Location.SourceSpan.End)
+            .SelectAsArray(d => (ObjectCreationExpressionSyntax)d.AdditionalLocations[0].FindNode(getInnermostNodeForTie: true, cancellationToken));
 
-            return Task.CompletedTask;
-        }
+#if CODE_STYLE
+        var options = CSharpSimplifierOptions.Default;
+#else
+        var options = (CSharpSimplifierOptions)await document.GetSimplifierOptionsAsync(fallbackOptions, cancellationToken).ConfigureAwait(false);
+#endif
 
-        private static void FixOne(SyntaxEditor editor, Diagnostic diagnostic, CancellationToken cancellationToken)
-        {
-            var node = diagnostic.AdditionalLocations[0].FindNode(getInnermostNodeForTie: true, cancellationToken);
-            editor.ReplaceNode(node, (current, _) =>
-            {
-                var currentObjectCreation = (ObjectCreationExpressionSyntax)current;
-                return SyntaxFactory.ImplicitObjectCreationExpression(
-                    WithoutTrailingWhitespace(currentObjectCreation.NewKeyword),
-                    currentObjectCreation.ArgumentList ?? SyntaxFactory.ArgumentList(),
-                    currentObjectCreation.Initializer);
-            });
-        }
+        await editor.ApplyExpressionLevelSemanticEditsAsync(
+            document,
+            nodes,
+            (semanticModel, node) => Analyze(semanticModel, options, node, cancellationToken),
+            (semanticModel, root, node) => FixOne(root, node),
+            cancellationToken).ConfigureAwait(false);
```

Comment:
needed to move to this form as the change from one fix might prevent another fix from applying.